packages/reactive-element/src/reactive-element.ts

```diff
@@ -1251,7 +1271,26 @@ export abstract class ReactiveElement
       ).getPropertyOptions(name);
       const hasChanged = options.hasChanged ?? notEqual;
       const newValue = this[name as keyof this];
-      if (hasChanged(newValue, oldValue)) {
+      let changed = hasChanged(newValue, oldValue);
+      // When there is no change, check a corner case that can occur when
+      // 1. there's a initial value which was not reflected
+      // 2. the property is subsequently set to this value.
+      // For example, `prop: {useDefault: true, reflect: true}`
+      // and el.prop = 'foo'. This should be considered a change if the
+      // attribute is not set because we will now reflect the property to the attribute.
+      if (
+        !changed &&
+        options.reflect &&
+        newValue != null &&
```

Comment:
Don't we need to run through the attribute converter to know if the property will be reflected? ie, if the converter returns `null`?

Reply:
Not sure we should bother since it's even more code? It's "correct" either way since the reflection will only happen if it should. In that rare case that it wouldn't reflect, we'd skip an update.